soc/wch/ch32v00x/Kconfig.defconfig

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2024 Michael Hope
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_CH32V003
+
+config NUM_IRQS
+	default 48
+
+config SYS_CLOCK_HW_CYCLES_PER_SEC
+	default 48000000
+
+config MAIN_STACK_SIZE
+	default 512
+
+config IDLE_STACK_SIZE
+	default 256
+
+config ISR_STACK_SIZE
+	default 256
+
+config CLOCK_CONTROL
+	default y
+
+config PINCTRL
+	default y
```

Comment:
Not allowed, drivers must select this

Reply:
Roger.

A bunch of existing code is doing the same thing though.

```C
$ cat soc/atmel/sam/Kconfig.defconfig
# Atmel SAM MCU family default configuration options

# Copyright (c) 2016 Piotr Mienkowski
# Copyright (c) 2024 Gerson Fernando Budke <nandojve@gmail.com>
# SPDX-License-Identifier: Apache-2.0

if SOC_FAMILY_ATMEL_SAM

rsource "*/Kconfig.defconfig"

config CLOCK_CONTROL
	default y

config GPIO
	default y

config PINCTRL
	default y

config SYS_CLOCK_HW_CYCLES_PER_SEC
	default $(dt_node_int_prop_int,/cpus/cpu@0,clock-frequency)

config WATCHDOG
	default y

endif # SOC_FAMILY_ATMEL_SAM
```

Also there are other examples:

```
$ rg "config PINCTRL" | grep soc     
soc/infineon/cat3/Kconfig.defconfig:config PINCTRL
soc/nxp/imxrt/Kconfig.defconfig:config PINCTRL_IMX
soc/microchip/mec/mec175x/Kconfig.defconfig.mec1753qlj:config PINCTRL
soc/microchip/mec/mec175x/Kconfig.defconfig.mec1753qsz:config PINCTRL
soc/microchip/mec/mec174x/Kconfig.defconfig.mec1743qsz:config PINCTRL
soc/microchip/mec/mec174x/Kconfig.defconfig.mec1743qlj:config PINCTRL
soc/microchip/mec/mech172x/Kconfig.defconfig.mech1723nlj:config PINCTRL
soc/microchip/mec/mech172x/Kconfig.defconfig.mech1723nsz:config PINCTRL
soc/gd/gd32/Kconfig.defconfig:config PINCTRL
soc/nxp/imx/imx8m/Kconfig.defconfig.mimx8ml8_a53:config PINCTRL_IMX
soc/nxp/imx/imx8m/Kconfig.defconfig.mimx8ml8_m7:config PINCTRL_IMX
soc/nxp/imx/imx8m/Kconfig.defconfig.mimx8mq6_m4:config PINCTRL_IMX
soc/nxp/imx/imx8m/Kconfig.defconfig.mimx8mn6_a53:config PINCTRL_IMX
soc/nxp/imx/imx8m/Kconfig.defconfig.mimx8mm6_m4:config PINCTRL_IMX
soc/nxp/imx/imx8m/Kconfig.defconfig.mimx8mm6_a53:config PINCTRL_IMX
soc/nxp/imx/imx7d/Kconfig.defconfig:config PINCTRL_IMX
soc/nxp/imx/imx6sx/Kconfig.defconfig:config PINCTRL_IMX
soc/nxp/imx/imx9/imx93/Kconfig.defconfig.mimx93.a55:config PINCTRL_IMX
soc/ti/k3/am6x/Kconfig.defconfig:config PINCTRL
soc/ite/ec/it8xxx2/Kconfig.defconfig.series:config PINCTRL
soc/atmel/sam/Kconfig.defconfig:config PINCTRL
soc/atmel/sam0/Kconfig.defconfig:config PINCTRL
soc/wch/ch32v00x/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra6m5/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra6e2/Kconfig.defconfig:config PINCTRL
soc/raspberrypi/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra4m3/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra8m1/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra6e1/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra2a1/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra8d1/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra6m3/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra6m2/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra8t1/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra6m1/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra4w1/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra4e2/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra6m4/Kconfig.defconfig:config PINCTRL
soc/renesas/ra/ra4m2/Kconfig.defconfig:config PINCTRL
soc/telink/tlsr/tlsr951x/Kconfig.defconfig:config PINCTRL
soc/renesas/rcar/rcar_gen4/Kconfig.defconfig.r8a779f0:config PINCTRL
soc/renesas/rcar/rcar_gen3/Kconfig.defconfig:config PINCTRL
```

Reply:
@nordicjm - can you suggest some changes that @kholia could implement that will unblock this? It looks like he has been waiting for your input for a couple of weeks.

Reply:
@nordicjm - simply repeating the same thing provides zero information gain to the author (i.e. does not help to clarify things).

I added a code suggestion on the USART driver. Can you confirm that the two lines here selecting PINCTRL should be removed in favour of one line selecting it in the USART driver?

Reply:
few PRs recently fixing this for other platforms, https://github.com/zephyrproject-rtos/zephyr/pull/80726 is an example, this should do the same

Reply:
@kholia I'll grab this and send you a PR

Reply:
https://github.com/nzmichaelh/zephyr/commit/6d4d98b5bd7816a3d427321030b65200d969647d removes the default and makes the USART driver select PINCTRL.

Reply:
Applied and pushed.